rust/lance-datafusion/src/planner.rs

```diff
@@ -665,6 +685,67 @@ impl Planner {
                 expr: Box::new(self.parse_sql_expr(expr)?),
                 data_type: self.parse_type(data_type)?,
             })),
+            SQLExpr::MapAccess { column, keys } => {
+                let mut expr = self.parse_sql_expr(column)?;
+
+                for key in keys {
+                    let field_access = match &key.key {
+                        SQLExpr::Value(
+                            Value::SingleQuotedString(s) | Value::DoubleQuotedString(s),
+                        ) => GetFieldAccess::NamedStructField {
+                            name: ScalarValue::from(s.as_str()),
+                        },
+                        SQLExpr::JsonAccess { .. } => {
+                            return Err(Error::invalid_input(
+                                "JSON access is not supported",
+                                location!(),
+                            ));
+                        }
+                        key => {
+                            let key = Box::new(self.parse_sql_expr(key)?);
+                            GetFieldAccess::ListIndex { key }
+                        }
+                    };
+
+                    let field_access_expr = RawFieldAccessExpr { expr, field_access };
+
+                    expr = self.plan_field_access(field_access_expr)?;
+                }
+
+                Ok(expr)
+            }
+            SQLExpr::Subscript { expr, subscript } => {
```

Comment:
Somewhat annoying that both `Subscript` and `MapAccess` exist :confused:

Reply:
Yeah and which path is triggered depends on the dialect. Implemented both for now.